### DISCUSSION TITLE
NTRU in strongSwan is outdated (how to use the last NTRU submission)

### DISCUSSION THREAD

**Discussion by Gor1234567** [strongswan/strongswan#1912] [General]
Hi! Overall, I want to use actual parameter sets in strongSwan 5.9.5 (for example, ntruhps4096821).

Using IKEv2 cipher suite, I try to connect strongSwan to another VPN, where I use NTRU cryptosystem, derived from the second round NIST submission. As I see, only the first round submission to NIST, called NTRUEncrypt, is realized in strongSwan. The problem is that it contains only old parameter sets which do not provide perfect correctness. 

Is there any NTRU param_set from the second round NIST Submission available (ntruhps2048509, ntruhps2048677, ntruhps4096821, ntruhrss701)? Do you plan to maintain them?

How did you check NTRU compatibility with other IPsec-based VPN solutions? Could you please name those solutions, which also use NTRU key exchange?

Best regards

**Comment by tobiasbrunner** [comment]
The _ntru_ plugin is deprecated and will be removed with the 6.0 release. You can look at the 6.0 beta releases (or the _six-beta_ branch) where the _oqs_ plugin (based on liboqs) supports newer NTRU versions (since NTRU was not selected by NIST eventually, it might not be in the final release as it's unlikely IKEv2 identifiers will be allocated for it).

**Reply by Gor1234567** [reply]
Thank you for the reply!

Is there any info about NTRU compatibility with other IPsec-based VPN solutions with whom you have tested it? I mean swans like strongSwan.

Also, seems like CRYSTALS-Kyber is a perfect choice for IKE, isn't it?

**Reply by tobiasbrunner** [reply]
> Is there any info about NTRU compatibility with other IPsec-based VPN solutions with whom you have tested it?

Not really, since there never was a standardized identifier. We tested against [ELVIS-PLUS](http://ipsec.elvis.ru/en.html) at some point, but the identifiers in the current beta are different and we've actually removed support for NTRU completely.

> Also, seems like CRYSTALS-Kyber is a perfect choice for IKE, isn't it?

Sure, it works fine. But has also not yet been standardized via IETF for use with IKEv2.

**Reply by Gor1234567** [reply]
Are there any updates in your Masterarbeit "Towards a Verifiably Secure Quantum-Resistant Key Exchange in IKEv2" in the paragraph "3.2 IPsec Protocol Suite"?

**Reply by tobiasbrunner** [reply]
That's not my master thesis and I don't know what exactly you are asking.

**Reply by Gor1234567** [reply]
O, sorry, it's a typo.

**Comment by Gor1234567** [comment]
According to the NIST IR 8413 Third Round Status Report, NTRU and FrodoKEM are KEMs no longer being considered.
What is the reason to change NTRU to FrodoKEM?

**Reply by strongX509** [reply]
FrodoKEM is recommended by the German BSI (Bundesamt für Sicherheit in der Informationstechnik) and will most probably become a BSI Standard. This might be the reason FrodoKEM has been retained in version 0.8 of the liboqs library although FrodoKEM is not a NIST Round 4 candidate.
